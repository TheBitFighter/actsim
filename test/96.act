deftype struct (bool x, y; int w) { }

defproc test()
{
   struct s[100];
   int i;
   bool qrst;

   chp {
      i := 0;
      s[0].w := 1;
      s[0].x := false;
      s[1].x := false;
      log ("got ", s[0].x, " ", s[0].w);
      log ("got ", s[1].x);
      s[i].x := true;
      s[0].w := s[0].w + 2;
      s[i+1].x := true; qrst := true;
      log ("got ", s[0].x, " ", s[0].w);
      log ("got ", s[1].x)
   }
}
