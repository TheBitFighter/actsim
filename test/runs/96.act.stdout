[                  40] <>  got 0 1
[                  40] <>  got 0
[                  80] <>  got 1 3
[                  80] <>  got 1
